### PR TITLE
Add integration tests across the compatibility matrix

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Add integration tests across the compatibility matrix: 1.3.6, 2.3.0

### Issues Resolved
Closes https://github.com/opensearch-project/spring-data-opensearch/issues/12

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -337,6 +338,23 @@
 			</resource>
 		</resources>
 
+		<testResources>
+			<testResource>
+				<directory>src/test/resources</directory>
+				<filtering>true</filtering>
+				<includes>
+					<include>**/testcontainers-opensearch.properties</include>
```
Comment (dlvenable) [inline pom.xml:346]: A newcomer to the project may not find it immediately obvious that this file is filtered. Might it make sense to have a directory within `resources` which is for filtering? e.g. `resources/filtered`. Also, future files that need to be filtered won't require modifications to these two lines.

Comment (reta) [reply pom.xml:346]: Property files are filtered very often, or at least this is normal even for newcomers, personally I would keep it in the root (and it is aligned with `src/main/resources` conventions)

**Comment by reta** [comment]
Interesting, 1 test case failure in 1.3.6 because of NPE in `GetResult::fromXContentEmbedded`: attempt to convert `null` to boolean. I will try to reproduce it in OpenSearch / 1.x.

**Comment by dblock** [comment]
I would split up an integration test matrix ala https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-integration.yml to make it faster/clearer and easy to extend.

**Comment by reta** [comment]
> I would split up an integration test matrix ala https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-integration.yml to make it faster/clearer and easy to extend.

Sure, will do that, thanks @dblock !

**Comment by reta** [comment]
@dblock created https://github.com/opensearch-project/OpenSearch/issues/4749, added conditional support for test cases to allow OpenSearch server version conditions .

**Review by dblock — APPROVED** [review]

**Pull request merged**